### ISSUE TITLE
Padding Mismatches Output Dimension in Conv1d

### ISSUE THREAD

**Issue by yhs0602** [descriptinc/descript-audio-codec#56]
Thank you for sharing your nice neural feature extractor of arbitrary audios!

In the source code (the `main` branch), there are several parts that mention`padding=1`, or etc. However, when I print out the dimensions of the outputs and the model itself, every Conv1d layer's `padding` parameter is not specified in the logs.

```text
Residual unit: 44532 -> 44514 by self.block=Sequential(
  (0): Snake1d()
  (1): Conv1d(64, 64, kernel_size=(7,), stride=(1,), dilation=(3,))
  (2): Snake1d()
  (3): Conv1d(64, 64, kernel_size=(1,), stride=(1,))
)
```

It seems to me that the `padding` parameters are being ignored?

I found out this when I tried to make an efficient function to map the original audio offset and the encoded code index.

```python
import math

from audiotools import AudioSignal

import dac


def get_conv1d_output_length(
    kernel_size,
    input_length,
    dilation=1,
    stride=1,
    padding=0,
):
    L = input_length
    d = dilation
    k = kernel_size
    s = stride
    p = padding
    L = math.floor(((L + 2 * p - d * (k - 1) - 1) / s) + 1)
    return L


def get_conv1d_transpose_output_length(dilation, kernel_size, stride, input_length):
    L = input_length
    d = dilation
    k = kernel_size
    s = stride
    L = (L - 1) * s + d * (k - 1) + 1
    return L


def get_residual_output_length(dilation, input_length):
    pad = 0  # ((7 - 1) * dilation) // 2
    # print("Main pad = ", pad)
    l_y = get_conv1d_output_length(
        kernel_size=7,
        dilation=dilation,
        padding=pad,
        input_length=input_length,
    )
    l_y = get_conv1d_output_length(
        kernel_size=1,
        input_length=l_y,
    )
    pad = (input_length - l_y) // 2
    if pad > 0:
        print(f"pad pad {pad=}")
        return input_length - 2 * pad
    print(f"==Residual: {input_length} -> {l_y}")
    return l_y


def get_encoder_block_output_length(stride, input_length):
    l = get_residual_output_length(dilation=1, input_length=input_length)
    l = get_residual_output_length(dilation=3, input_length=l)
    l = get_residual_output_length(dilation=9, input_length=l)
    l = get_conv1d_output_length(
        kernel_size=2 * stride,
        stride=stride,
        padding=0,  # math.ceil(stride / 2),
        input_length=l,
    )
    print(f"=Encoder block:  {input_length} -> {l}")
    return l


def get_encoded_dim(input_length):
    # wnconv1d k=7 p=3 s=1 d=1
    l = get_conv1d_output_length(
        kernel_size=7, padding=0, input_length=input_length
    )  # padding=3
    print(f"After encoder first conf1d: {input_length} -> {l=}")
    # EncoderBlock(stride=2, 4, 8, 8)
    # EncoderBlock = Residual(dilation=1, 3, 9), WNConv1d(k=2*stride, stride=stride, padding=ceil(stride/2))
    l = get_encoder_block_output_length(2, l)
    l = get_encoder_block_output_length(4, l)
    l = get_encoder_block_output_length(8, l)
    l = get_encoder_block_output_length(8, l)
    l = get_conv1d_output_length(
        kernel_size=3, padding=0, input_length=l
    )  # padding = 1
    return l


def main():
    model_path = dac.utils.download(model_type="44khz")
    model = dac.DAC.load(model_path)
    model.to("mps:0")
    signal = AudioSignal("kyunggi.wav")
    signal.to_mono()
    x = model.compress(signal)  # <class 'dac.model.base.DACFile'>
    # print(f"{type(x)=}")
    print(f"{x.codes.shape=}")
    print(f"{x.codes=}")
    print(f"{x.chunk_length=}")
    print(f"{x.original_length=}")
    print(f"{x.sample_rate=}")
    print(f"{signal.signal_length=} {signal.signal_duration=}")
    print(f"{model.get_output_length(403956)=}")
    # x.codes.shape[2] = 1080
    # x.chunk_length = 72
    # x.original_length = 403956
    # x.sample_rate = 44100
    # signal.signal_length = 403956
    # signal.signal_duration = 9.15
    # 403956/44100 = 9.15s
    # signal_duration = signal_length / sample_rate
    # Save and load to and from disk
    # x.save("compressed.dac")
    # x.original_length = 623616 # 668,160 403956
    # y = model.decompress(x)


if __name__ == "__main__":
    print(f"{get_encoded_dim(44544)=}")  # 44538 44544
    main()
```

The code above calculates 72 codes, given 44544 samples, which is the same as the output of the encoder. However, I had to set all the paddings to 0 when calculating, to match the result. I mean, I could not find any way to match the output dimension using the padding values specified in the source code.

**Comment by yhs0602** [comment]
```
    @padding.setter
    def padding(self, value):
        assert isinstance(value, bool)

        layers = [
            l for l in self.modules() if isinstance(l, (nn.Conv1d, nn.ConvTranspose1d))
        ]

        for layer in layers:
            if value:
                if hasattr(layer, "original_padding"):
                    layer.padding = layer.original_padding
            else:
                layer.original_padding = layer.padding
                layer.padding = tuple(0 for _ in range(len(layer.padding)))

        self._padding = value
```

Seems like this code enables/disables the padding values of the conv1d layers, and `compress` method do change this value.
